File: uhrwerk-backend-mysql/src/test/java/io/qimia/uhrwerk/dao/HashIdTest.java
package io.qimia.uhrwerk.dao;

import io.qimia.uhrwerk.common.model.Dependency;
import net.openhft.hashing.LongHashFunction;
import org.apache.log4j.Logger;
import org.junit.jupiter.api.Test;

public class HashIdTest {
  private final Logger logger = Logger.getLogger(this.getClass());

  @Test
  void test() {
    Dependency dep = new Dependency();
    dep.setArea("test-area");
    dep.setVertical("test-vertical");
    dep.setTableName("test-table");
    dep.setFormat("json");
    dep.setVersion("version");
    StringBuilder res =
        new StringBuilder()
            .append(dep.getArea())
            .append(dep.getVertical())
            .append(dep.getTableName())
            .append(dep.getVersion());
    logger.info("without format: " + res.toString());
    long tableId = LongHashFunction.xx().hashChars(res);
    logger.info("tableId: " + tableId);
    res = res.append(dep.getFormat());
    logger.info("with format: " + res.toString());
    long depId = LongHashFunction.xx().hashChars(res);
    logger.info("depId: " + depId);
  }
}

File: uhrwerk-backend-mysql/src/test/kotlin/io/qimia/uhrwerk/dao/HashIdTest.kt
package io.qimia.uhrwerk.dao

import io.qimia.uhrwerk.common.metastore.model.DependencyModel
import net.openhft.hashing.LongHashFunction
import org.junit.jupiter.api.Test
import org.slf4j.LoggerFactory

class HashIdTest {
    private val logger = LoggerFactory.getLogger(this.javaClass)
    @Test
    fun test() {
        val dep = DependencyModel()
        dep.area = "test-area"
        dep.vertical = "test-vertical"
        dep.tableName = "test-table"
        dep.format = "json"
        dep.version = "version"
        var res = StringBuilder()
                .append(dep.area)
                .append(dep.vertical)
                .append(dep.tableName)
                .append(dep.version)
        logger.info("without format: $res")
        val tableId = LongHashFunction.xx().hashChars(res)
        logger.info("tableId: $tableId")
        res = res.append(dep.format)
        logger.info("with format: $res")
        val depId = LongHashFunction.xx().hashChars(res)
        logger.info("depId: $depId")
    }
}